crates/sui-framework/packages/dwallet-system/sources/sui_state_proof.move

```diff
@@ -0,0 +1,145 @@
+        committee: vector<u8>,
+    }
+
+
+    struct EpochCommitteeSubmitted has copy, drop {
+        epoch: u64,
+        epoch_committee_id: ID,
+    }
+
+    struct CapWrapper has key, store {
+        id: UID,
+        cap_id_sui: ID,
+        cap: DWalletCap,
+    }
+
+
+    // TODO add wittness here so it can be initialized only once??
+    public fun init_module(init_committee: vector<u8>, package_id: vector<u8>, event_type_layout: vector<u8>, message_field_name: vector<u8>, ctx: &mut TxContext) {
+        let registry = StateProofRegistry {
+            id: object::new(ctx),
+            highest_epoch: 0,
+        };
+
+        let config = StateProofConfig {
+            id: object::new(ctx),
+            registry_id: object::id(&registry),
+            package_id: package_id,
+            event_type_layout: event_type_layout,
+            message_field_name: message_field_name,
+        };
+
+        let first_committee = EpochCommittee {
+            id: object::new(ctx),
+            committee: init_committee,
+        };
+
+        event::emit(EpochCommitteeSubmitted {
+            epoch: 0,
+            epoch_committee_id: object::id(&first_committee),
+        });
+        
+        transfer::share_object(registry);
+        transfer::freeze_object(config);
+        transfer::freeze_object(first_committee);
+    }
+
+    
+
+    native fun sui_state_proof_verify_committee(prev_committee: vector<u8>, checkpoint_summary: vector<u8>): vector<u8>;
+
+
+    public fun submit_new_state_committee(
+        registry: &mut StateProofRegistry,
+        prev_committee: &EpochCommittee,
+        new_checkpoint_summary: vector<u8>,
+        ctx: &mut TxContext,
+    ) {
+        let committee_verified_bytes = sui_state_proof_verify_committee(prev_committee.committee, new_checkpoint_summary);
+
+        let committee_new = EpochCommittee {
+                                    id: object::new(ctx),
+                                    committee: committee_verified_bytes,
+                                    };
+
+        registry.highest_epoch = registry.highest_epoch + 1;
+
+        event::emit(EpochCommitteeSubmitted {
+            epoch: registry.highest_epoch,
+            epoch_committee_id: object::id(&committee_new),
+        });
+        
+        transfer::freeze_object(committee_new);
+    }
+
+
+    public fun create_dwallet_wrapper(
+        config: &StateProofConfig,
+        dwallet_cap: DWalletCap, 
+        committee: &EpochCommittee,
+        checkpoint_summary: vector<u8>,
+        checkpoint_contents: vector<u8>,
+        transaction: vector<u8>,
+        ctx: &mut TxContext
+    ){        
+        
+        let (cap_id_bytes, _) = sui_state_proof_verify_transaction(committee.committee, checkpoint_summary, checkpoint_contents, transaction, config.event_type_layout, config.package_id );
+        let wrapper = CapWrapper {
+            id: object::new(ctx),
+            cap_id_sui: object::id_from_bytes(cap_id_bytes),
```

Comment:
Notice that we are calling the `object::id_from_bytes()` function, that takes a *sui* serialized object, and attempts to parse it using the *dwallet network* code. 

This works now because we are a sui fork, and this structure hasn't changed in the version differences.

However, this is the wrong approach.

Instead, we should somehow import the *sui* `object::id_from_bytes` function and call it.
We should also pay attention to version updates to see it won't be changed.

Alternatively, we should be in touch with the sui team to assure this object structure will never be changed.

In any case this needs proper thinking